### PR TITLE
feat: add WithMalformedRowPolicy for ragged CSV/TSV rows

### PR DESCRIPTION
**Pull request by nao1215**
## Summary

A CSV/TSV row whose field count differs from the header was reported by `encoding/csv` as a field-count error, which the streaming loader masked by creating an empty single-column table without surfacing the error. Both the malformed row and the well-formed rows read before it were silently dropped.

This adds `WithMalformedRowPolicy` so callers choose how a ragged row is handled:

- `MalformedRowStop` (default): abort the import with `ErrColumnMismatch` instead of masking the failure with an empty table.
- `MalformedRowSkip`: drop ragged rows and keep the well-formed ones.
- `MalformedRowFill`: pad short rows with empty strings and truncate long rows to the header width.

The delimited reader now sets `FieldsPerRecord = -1` and reconciles the field count itself, and the streaming loader propagates a terminal parse error instead of degrading to an empty table. The policy applies to delimited text only; XLSX, LTSV, Parquet, and JSON/JSONL have no per-row field-count mismatch.

## Behavior change

The default policy is stop, so a ragged CSV/TSV now fails the import with `ErrColumnMismatch` rather than producing an empty table. One existing edge-case test that asserted the old silent behavior for inconsistent row lengths was updated to expect the error.

## Tests

- New `malformed_row_test.go` covers stop/skip/fill across CSV and TSV, short and long rows, the default policy, and a metamorphic check that a well-formed file imports identically under every policy.
- Full suite and golangci-lint are green.

Ref: https://github.com/nao1215/sqly/issues/731

## Summary by CodeRabbit

* **New Features**
  * Added a new option for CSV/TSV imports to control how rows with too few or too many fields are handled: stop, skip, or fill.
  * You can now choose whether malformed rows abort the import, are ignored, or are adjusted to match the header.

* **Bug Fixes**
  * Fixed an issue where ragged CSV/TSV rows could be silently dropped and an empty table produced.
  * The default behavior now stops the import with an error when column counts do not match.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+- **`WithMalformedRowPolicy` lets callers choose how a ragged CSV/TSV row is handled.** A row whose field count differs from the header can now stop the import (`MalformedRowStop`, the default), be dropped (`MalformedRowSkip`), or be reshaped to the header width by padding short rows with empty strings and truncating long rows (`MalformedRowFill`). The policy applies to delimited text only; XLSX, LTSV, Parquet, and JSON/JSONL have no per-row field-count mismatch. (Ref [nao1215/sqly#731](https://github.com/nao1215/sqly/issues/731))
+
+### Fixed
+- **A ragged CSV/TSV row no longer silently drops data.** A row with fewer or more fields than the header made `encoding/csv` return a field-count error, which the streaming loader masked by creating an empty single-column table, losing both the malformed row and the well-formed rows before it. The default policy now aborts the import with `ErrColumnMismatch` instead of producing an empty table. (Ref [nao1215/sqly#731](https://github.com/nao1215/sqly/issues/731))
+
```
[the remark this answers is not in the document]
Comment (nao1215) [reply CHANGELOG.md:15]: Addressed in fdec42b: both entries now link this PR (#145) and the commit, keeping the sqly issue as context.
